apps/givenergysolar/givenergy_solar.star

```diff
@@ -0,0 +1,201 @@
+            dir = 0
+        points.append((IMG, o["solar"]["power"] / 1000))
+        flows.append(dir)
+
+        o["grid"]["power"] = o["grid"]["power"] * -1
+
+        points.append((HOUSE, o["consumption"] / 1000))
+        points.append((GRID, o["grid"]["power"] / 1000))
+
+        if o["grid"]["power"] < -10:
+            dir = 1
+        elif o["grid"]["power"] > 10:
+            dir = -1
+        else:
+            dir = 0
+
+        flows.append(dir)
+
+        columns = []
+        for p in points:
+            IMG, power = p
+            columns.append(
+                render.Column(
+                    main_align = "space_between",
+                    cross_align = "center",
+                    children = [
+                        render.Image(src = IMG),
+                        render.Text(
+                            content = format_power(power),
+                            height = 8,
+                            font = "tb-8",
+                            color = "#ffd11a",
+                        ),
+                        render.Text(
+                            content = unit,
+                            height = 8,
+                            font = "tb-8",
+                            color = "#ffd11a",
+                        ),
+                    ],
+                ),
+            )
+        dots = [render.Box(width = 18)]
+        for dir in flows:
+            if dir:
+                el = render.Image(src = DOTS_LTR if dir == 1 else DOTS_RTL)
+            else:
+                el = render.Box(width = 10)
+            dots.append(
+                render.Stack(children = [render.Box(width = 21), el]),
+            )
+        return render.Root(
+            child = render.Stack(
+                children = [
+                    render.Row(
+                        expanded = True,
+                        main_align = "space_between",
+                        children = columns,
+                    ),
+                    render.Column(
+                        children = [
+                            render.Box(height = 8),
+                            render.Row(expanded = True, children = dots),
+                        ],
+                    ),
+                ],
+            ),
+        )
+    else:
+        error_message = error_details["error_section"] + ". " + error_details["error"]
+        return render.Root(
+            child = render.Marquee(
+                width = 64,
+                child = render.Text(error_message),
+            ),
+        )
+
+def format_power(p):
+    if p:
+        return humanize.float("#,###.##", p)
+    else:
+        return "0"
+
+def get_schema():
+    return schema.Schema(
+        version = "1",
+        fields = [
+            schema.Text(
+                id = "api_key",
+                name = "GivEnergy API Key",
+                desc = "GivEnergy API Key, available from https://givenergy.cloud/account-settings/security",
```

Comment:
Note that the field descriptions are currently not visible to users in the iOS/Android app, so this might need to be mentioned in the app description in the manifest file.